### PR TITLE
ROX-9115: Add suite/case/config/debug options to e2e test runner

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

Continues from #3888 to:
- Allow `qa-tests-backend/` tests to run just for a suite or suite and case.
- Run just the 'configuration' steps for `qa-tests-backend/` i.e. setup and deployment.
- Support post test failure debug gathering.
- Examples.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

CI+:
- [x] openshift and upgrade test for `QA_TEST_DEBUG_LOGS` env var dependency coverage

Manual:
- [x] debug is gathered when a test fails with `-d`
- [x] logs still gathered on failure in CI

### CODE REVIEW

File: qa-tests-backend/src/main/groovy/util/Helpers.groovy [hunk 1]
```diff
@@ -128,15 +133,20 @@ class Helpers {
 
     // collectImageScanForDebug(image) - a best effort debug tool to get a complete image scan.
     static void collectImageScanForDebug(String image, String saveName) {
-        if (!Env.IN_CI) {
-            log.info "Won't collect image scans when not in CI"
+        if (!Env.IN_CI && !Env.GATHER_QA_TEST_DEBUG_LOGS) {
+            log.info "Won't collect image scans without CI=true or GATHER_QA_TEST_DEBUG_LOGS=true"
```
Comment (janisz) [inline qa-tests-backend/src/main/groovy/util/Helpers.groovy:137]:
```suggestion
            log.info "Won't collect image scans without CI=true and GATHER_QA_TEST_DEBUG_LOGS=true"
```

File: qa-tests-backend/src/main/groovy/util/Helpers.groovy [hunk 2]
```diff
@@ -88,8 +88,13 @@ class Helpers {
     }
 
     static void collectDebugForFailure(Throwable exception) {
-        if (!Env.IN_CI) {
-            log.info "Won't collect logs when not in CI"
+        if (!Env.IN_CI && !Env.GATHER_QA_TEST_DEBUG_LOGS) {
+            log.info "Won't collect logs without CI=true or GATHER_QA_TEST_DEBUG_LOGS=true"
```
Comment (janisz) [inline qa-tests-backend/src/main/groovy/util/Helpers.groovy:92]:
```suggestion
            log.info "Won't collect logs without CI=true and GATHER_QA_TEST_DEBUG_LOGS=true"
```

File: qa-tests-backend/src/main/groovy/util/Helpers.groovy [hunk 3]
```diff
@@ -128,15 +133,20 @@ class Helpers {
...
     static void collectImageScanForDebug(String image, String saveName) {
-        if (!Env.IN_CI) {
-            log.info "Won't collect image scans when not in CI"
+        if (!Env.IN_CI && !Env.GATHER_QA_TEST_DEBUG_LOGS) {
+            log.info "Won't collect image scans without CI=true or GATHER_QA_TEST_DEBUG_LOGS=true"
+            return
+        }
+
+        if (Env.QA_TEST_DEBUG_LOGS == "") {
+            log.info "Won't collect image scans when QA_TEST_DEBUG_LOGS is not set"
```
Comment (janisz) [inline qa-tests-backend/src/main/groovy/util/Helpers.groovy:142]: This conditions are in two places how about extracting them to a function that will return bool if logs should be collected?

Comment (gavin-stackrox) [reply qa-tests-backend/src/main/groovy/util/Helpers.groovy]: Yeah, I'll do that.

File: tests/e2e/run-e2e-tests.sh [hunk 4]
```diff
@@ -10,24 +10,51 @@ ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../.. && pwd)"
...
+source "$ROOT/tests/e2e/lib.sh"
 
-usage() { 
+require_environment "QA_TEST_DEBUG_LOGS"
+
+usage() {
+    script=$(basename "$0")
     cat <<_EOH_
-Usage: $0 [-m <tag>] [-o k8s|openshift] [-y] [<e2e flavor: one of qa|e2e|ui|upgrade, defaults to qa>]
+Usage: $script [Options...] [E2e flavor] [Suite] [Case]
```
Comment (porridge) [inline tests/e2e/run-e2e-tests.sh:20]:
```suggestion
Usage:
 $script [Options...] [E2e flavor]
 
   Configures the cluster and runs all suites.
 
 $script [Options...] [E2e flavor] Suite [Case]
 
   Expects a previously configured cluster and runs only selected suite/case.
 
```

**Review by porridge — APPROVED** [review]

File: qa-tests-backend/src/main/groovy/util/Helpers.groovy [hunk 5]
```diff
@@ -163,4 +161,17 @@ class Helpers {
         proc.waitFor()
         log.debug "Ran: ${cmd}\nExit: ${proc.exitValue()}\nStdout: $sout\nStderr: $serr"
     }
+
+    private static Boolean collectDebug() {
```
Comment (janisz) [inline qa-tests-backend/src/main/groovy/util/Helpers.groovy:165]: nit
```suggestion
    private static boolean collectDebug() {
```

**Review by janisz — APPROVED** [review]

**Pull request merged**